crates/rpc/src/v04/method/trace_transaction.rs

```diff
@@ -0,0 +1,173 @@
+use anyhow::Context;
+use pathfinder_common::{BlockHash, BlockId, TransactionHash};
+use pathfinder_executor::{CallError, Transaction};
+use primitive_types::U256;
+use serde::{Deserialize, Serialize};
+use tokio::task::JoinError;
+
+use crate::{
+    compose_executor_transaction,
+    context::RpcContext,
+    error::{RpcError, TraceError},
+    executor::ExecutionStateError,
+};
+
+use super::simulate_transactions::dto::TransactionTrace;
+
+#[derive(Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct TraceTransactionInput {
+    pub transaction_hash: TransactionHash,
+}
+
+#[derive(Debug, Serialize, Eq, PartialEq)]
+pub struct TraceTransactionOutput(pub TransactionTrace);
+
+#[derive(Debug)]
+pub enum TraceTransactionError {
+    InvalidTxnHash,
+    NoTraceAvailable(TraceError),
+    Internal(anyhow::Error),
+}
+
+impl From<ExecutionStateError> for TraceTransactionError {
+    fn from(value: ExecutionStateError) -> Self {
+        match value {
+            ExecutionStateError::BlockNotFound => Self::NoTraceAvailable(TraceError::Rejected),
```

Comment:
According to the spec we should return `TraceError::Rejected` for rejected transactions -- these transactions are rejected right away on the gateway so they don't show up _at all_ in blocks so we don't have them.

`TraceError::Received` is also something we don't do -- we do on-request execution so there's no case it should be returned.

In which case do we get en ExecutionStateError::BlockNotFound? Maybe it's not even something we should be getting during normal operations?